crates/subspace-farmer/src/single_disk_farm/plot_cache.rs

```diff
@@ -87,39 +80,36 @@ impl DiskPlotCache {
             let file_size = sector_size * u64::from(target_sector_count);
             let plotted_size = sector_size * sectors_metadata.len() as u64;
 
-            // Avoid writing over large gaps on Windows that is very lengthy process
-            if !cfg!(windows) || (file_size - plotted_size) <= MAX_WINDOWS_PLOT_SPACE_FOR_CACHE {
-                // Step over all free potential offsets for pieces that could have been cached
-                let from_offset = (plotted_size / Self::element_size() as u64) as u32;
-                let to_offset = (file_size / Self::element_size() as u64) as u32;
-                // TODO: Parallelize or read in larger batches
-                for offset in (from_offset..to_offset).rev() {
-                    match Self::read_piece_internal(file, offset, &mut element) {
-                        Ok(maybe_piece_index) => match maybe_piece_index {
-                            Some(piece_index) => {
-                                map.insert(RecordKey::from(piece_index.to_multihash()), offset);
-                            }
-                            None => {
-                                next_offset.replace(offset);
-                                break;
-                            }
-                        },
-                        Err(DiskPlotCacheError::ChecksumMismatch) => {
-                            next_offset.replace(offset);
-                            break;
+            // Step over all free potential offsets for pieces that could have been cached
+            let from_offset = (plotted_size / Self::element_size() as u64) as u32;
```

Comment:
This conversion introduces an implicit constraint on file size. Is the u32 type necessary?